examples/gallery/embellishments/scalebar.py

```diff
@@ -0,0 +1,97 @@
+r"""
+Scale bar
+=========
+
+The ``map_scale`` parameter of the :meth:`pygmt.Figure.basemap` and
+:meth:`pygmt.Figure.coast` methods is used to add a scale bar to a map.
+This example shows how such a scale bar can be customized:
+
+ - position: **j**. Set the reference point by specifying a two-letter (order
+   independent) code, chosen from vertically **T**\(op), **M**\(iddle), or
+   **B**\(ottom) and horizontally **L**\(eft), **C**\(entre), or **R**\(ight).
+ - justify: **+j**. Set the anchor point by specifying a two-letter (order
+   independent) code, chosen from vertically **T**\(op), **M**\(iddle), or
+   **B**\(ottom) and horizontally **L**\(eft), **C**\(entre), or **R**\(ight).
+ - offset: **+o**\ *xoffset*/*yoffset*. Shift from ``position`` in x
+   (longitude) and y (latitude) directions.
+ - length: **+w**\ *length*. Give value and unit.
+ - height: :gmt-term:`MAP_SCALE_HEIGHT`
+ - origin on map: **+c**\ [*slon*/]\ *slat*. *slon* is only optional for
+   projections with constant scale along parallels.
+ - fancy style: **+f**. Get a scale bar that looks like train tracks.
+ - label: **+l**. Use another label as the unit given via **+w**.
+ - label alignment: **+a**. Append **t**\(op) [Default], **b**\(ottom),
+   **l**\(eft), or **r**\(ight).
+ - distance unit: **+u**. Add the distance unit to the distance values.
+
+Via the ``box`` parameter a box can be added around the scale bar.
+"""
+
+
+# %%
+import pygmt
+
+# Create a new Figure instance
+fig = pygmt.Figure()
+
+fig.basemap(
+    region=[-45, -25, -15, 0],
+    projection="M10c",  # Mercator projection with 10 centimeters width
+    frame=["WSne", "af"],
+    map_scale="jMC+w1000k",
+)
+
+fig.shift_origin(xshift="+w1c")
+
+fig.basemap(
+    region=[-45, -25, -15, 0],
+    projection="M10c",
+    frame=["wSnE", "af"],
+    map_scale="jML+jBL+o1c/1c+w1000k+f+u",
+)
+
+fig.show()
+
+# %%
+# Amazing text
```

Comment:
```suggestion
```
Is this only a wildcard?

Reply:
Yes, the documentation is not completed yet. I realized we need to add the alias `box` for **-F** of `pygmt.Figure.coast` first before continuing here. Then I will think in more detail about which modifiers of `box` make sense to be mentioned in this gallery example, as this example should be primarily about adding a scale bar itself.

Reply:
> Then I will think in more detail about which modifiers of `box` make sense to be mentioned in this gallery example, as this example should be primarily about adding a scale bar itself.

Yes, `box` is a parameter shared by many modules like `coast`, `image`, `logo`, so we should have a separate example for `box`.

Reply:
So, the PR for adding `box` as alias of **-F** for `pygmt.Figure.coast` was merged, and I adjusted code and docs. I think the currently mentioned modifiers (**+g**, **+p**, and **+r**) are sufficient for this gallery example:

https://github.com/GenericMappingTools/pygmt/blob/d8065edf811eaa33da34b518d38ed32904507eb4/examples/gallery/embellishments/scalebar.py#L116-L120
 
https://github.com/GenericMappingTools/pygmt/blob/d8065edf811eaa33da34b518d38ed32904507eb4/examples/gallery/embellishments/scalebar.py#L133-L136